File: DynamicProgramming/070.climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        
        if(n==0){
            return 0;
        }else if(n==1){
            return 1;
        }else if(n==2){
            return 2;
        }
        vector<int> stairs; 
        stairs.push_back(0);
        stairs.push_back(1);
        stairs.push_back(2);
        
        for (int i=3; i<n; i++){
            stairs.push_back(stairs[i-2] + stairs[i-1]);
        }
        return stairs[n-2] + stairs[n-1]; 
    }
};

File: DynamicProgramming/070.climbing-stairs.js
const climbStairs = (n) => {
    const stairs = [0, 1 ,2];
    for (let i = 3; i <= n; i++) {
        stairs[i] = stairs[i-2] + stairs[i-1];
    }
    return stairs[n];
}

console.log(climbStairs(5));